app/controllers/passwords_controller.rb

```diff
@@ -41,6 +47,9 @@ def webauthn_edit
     if params[:credentials].blank?
       login_failure(t("credentials_required"))
       return
+    elsif !session_active?
```

Comment:
This suggests an alternative to my note about about `elsif !session_active?`, which would be to instead define a `session_inactive?` method and use it in both places.

Reply:
I'm more inclined to keep it as `session_active?` due to the fact that we are not negating elsewhere. If we were only checking if the session is inactive, I would agree `session_inactive?` would be better. But, given that we also test for `session_active?` to be true, I think it's better to stay like this. 

This is because where we do `if session_active?`, that is saying `is the session active?`, where as `if !session_inactive?` is saying "is the session not not active?". Given that `session_inactive?` introduces a logical double negation, whereas `!session_active?` is only a single negation, I think it makes more sense to keep it as is.

I hope that makes sense, happy to discuss or clarify further.